commerce/domain/tax/cart/calculator/classes/TaxCartCalculatorSample.cls

```diff
@@ -1,349 +1,242 @@
+                taxData = getTaxesFromExternalService(
+                cartItemByIdMap, cartDeliveryGroup.getDeliverToAddress().getState(),
+                cartDeliveryGroup.getDeliverToAddress().getCountry(), cart.getTaxType());
+
+                taxDataShippingItems = getTaxesFromExternalService(
+                chargeItemByDeliveryGroupIdMap, cartDeliveryGroup.getDeliverToAddress().getState(),
+                cartDeliveryGroup.getDeliverToAddress().getCountry(), cart.getTaxType());
+
+            } else{
+                taxData = getTaxesFromStaticResponse(
+                cartItemByIdMap, cartDeliveryGroup.getDeliverToAddress().getState(),
+                cartDeliveryGroup.getDeliverToAddress().getCountry(), cart.getTaxType());
+
+                taxDataShippingItems = getTaxesFromStaticResponse(
+                chargeItemByDeliveryGroupIdMap, cartDeliveryGroup.getDeliverToAddress().getState(),
+                cartDeliveryGroup.getDeliverToAddress().getCountry(), cart.getTaxType());
             }
-        }
-      }
-
-      // If there are shipping items, add tax for them as well
-      for (String cartItemId : dataFromExternalServiceForShippingItems.keySet()) {
-        TaxDataFromExternalService taxDetailsToCartId = dataFromExternalServiceForShippingItems.get(cartItemId);
-        CartExtension.CartItem cartItem = shippingItemById.get(cartItemId);
-        boolean isCartItemModified = false;
-        // If there is any modification in unit price, delete existing and create new cart tax entries in cart item.
-        if (cartItem.getNetUnitPrice() != null &&
-          cartItem.getNetUnitPrice() != taxDetailsToCartId.getNetUnitPrice()) {
-            cartItem.getCartTaxes().remove(cartItem.getCartTaxes().get(0));
-            isCartItemModified = true;
-        }
 
-        if (cartItem.getCartTaxes() == null ||
-          cartItem.getCartTaxes().isEmpty() ||
-          isCartItemModified == true) {
-            cartItem.setNetUnitPrice(taxDetailsToCartId.getNetUnitPrice());
-            cartItem.setGrossUnitPrice(taxDetailsToCartId.getGrossUnitPrice());
-            CartExtension.CartTaxList cartTaxCollection = cartItem.getCartTaxes();
-            CartExtension.CartTax cartTax = new CartExtension.CartTax(
-              CartExtension.TaxTypeEnum.ESTIMATED,
-              taxDetailsToCartId.getAmount(),
-              taxDetailsToCartId.getTaxName()
-            );
-            cartTax.setTaxRate(String.valueOf(taxDetailsToCartId.getRate()));
-            cartTaxCollection.add(cartTax);
+            // TODO need to add cvo in case of tax data is null
+            if (taxData == null) {
+                CartExtension.CartValidationOutput cvo = new CartExtension.CartValidationOutput(
+                CartExtension.CartValidationOutputTypeEnum.TAXES,
+                CartExtension.CartValidationOutputLevelEnum.ERROR);
+                cvo.setMessage('No tax rates configured.');
```

Comment:
This is the message that would be populated to the Shopper. Not sure that makes a lot of sense and shopper can do anything about it.